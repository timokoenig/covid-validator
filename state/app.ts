import { entity, persistence } from 'simpler-state'

export type AppState = {
  country: string
  state: string
  purpose: string
  counter: number
  showCounter: boolean
}

export const app = entity<AppState>(
  { country: 'DE', state: '', purpose: '3G', counter: 0, showCounter: false },
  [persistence('app')]
)

export const setCountry = (country: string) => {
  app.set({ ...app.get(), country })
}

export const setState = (state: string) => {
  app.set({ ...app.get(), state })
}

export const setPurpose = (purpose: string) => {
  app.set({ ...app.get(), purpose })
}

export const toggleCounter = () => {
  const state = app.get()
  app.set({ ...state, showCounter: !state.showCounter })
}

export const resetCounter = () => {
  app.set({ ...app.get(), counter: 0 })
}

export const decreaseCounter = () => {
  const state = app.get()
  if (state.counter == 0) return
  app.set({ ...state, counter: state.counter - 1 })
}

export const increaseCounter = () => {
  const state = app.get()
  app.set({ ...state, counter: state.counter + 1 })
}
