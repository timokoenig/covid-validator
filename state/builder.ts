import { entity, persistence } from 'simpler-state'
import { CustomRule } from '../utils/certlogic'

export type BuilderState = {
  customRules: CustomRule[]
}

export const builder = entity<BuilderState>({ customRules: [] }, [persistence('builder')])

export const setRules = (customRules: CustomRule[]) => {
  builder.set({ ...builder.get(), customRules })
}
