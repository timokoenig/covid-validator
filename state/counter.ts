import { entity, persistence } from 'simpler-state'

export const counter = entity(0, [persistence('counter')])

export const reset = () => {
  counter.set(0)
}

export const decrease = () => {
  if (counter.get() == 0) return
  counter.set(counter.get() - 1)
}

export const increase = () => {
  counter.set(counter.get() + 1)
}
