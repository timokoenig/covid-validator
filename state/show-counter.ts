import { entity, persistence } from 'simpler-state'
import { reset } from './counter'

export const showCounter = entity(false, [persistence('show-counter')])

export const toggle = () => {
  reset()
  showCounter.set(!showCounter.get())
}
