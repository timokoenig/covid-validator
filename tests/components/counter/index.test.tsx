import React from 'react'
import renderer from 'react-test-renderer'
import { resetAll } from 'simpler-state'
import Counter from '../../../components/counter'
import { app } from '../../../state/app'

describe('Counter', () => {
  beforeEach(() => {
    resetAll()
  })

  it('renders correctly', () => {
    app.set({ ...app.get(), showCounter: true })

    const tree = renderer.create(<Counter />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with set counter', () => {
    app.set({ ...app.get(), counter: 1, showCounter: true })

    const tree = renderer.create(<Counter />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly without counter', () => {
    app.set({ ...app.get(), showCounter: false })

    const tree = renderer.create(<Counter />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
