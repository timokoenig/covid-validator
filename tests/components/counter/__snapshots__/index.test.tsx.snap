// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-qc2p2d{border-radius:lg;overflow:hidden;background:blue.400;box-shadow:xl;margin-bottom:10px;padding:5px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;vertical-align:center;}",
      }
    }
    data-emotion="css qc2p2d"
  />,
  <div
    className="css-qc2p2d"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-57pqds{-webkit-flex:1;-ms-flex:1;flex:1;vertical-align:center;}",
        }
      }
      data-emotion="css 57pqds"
    />
    <div
      className="css-57pqds"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-mmcpkq{font-weight:semibold;text-transform:uppercase;color:white;}",
          }
        }
        data-emotion="css mmcpkq"
      />
      <p
        className="chakra-text css-mmcpkq"
      >
        counter
      </p>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1j71ma5{font-size:2xl;color:white;}",
          }
        }
        data-emotion="css 1j71ma5"
      />
      <p
        className="chakra-text css-1j71ma5"
      >
        0
      </p>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-eg4v4b{margin-top:1px;}",
        }
      }
      data-emotion="css eg4v4b"
    />
    <div
      className="css-eg4v4b"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-c32o75{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;position:relative;white-space:nowrap;vertical-align:middle;outline:2px solid transparent;outline-offset:2px;width:auto;color:white;background-color:blue.400;}.css-c32o75:hover,.css-c32o75[data-hover]{background:blue.300;}.css-c32o75:active,.css-c32o75[data-active]{background:blue.400;}",
          }
        }
        data-emotion="css c32o75"
      />
      <button
        className="chakra-button css-c32o75"
        onClick={[Function]}
        type="button"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-19svxgr{width:5;height:5px;display:inline-block;line-height:1em;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;color:currentColor;vertical-align:middle;}",
            }
          }
          data-emotion="css 19svxgr"
        />
        <svg
          className="chakra-icon css-19svxgr"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <g
            fill="currentColor"
          >
            <rect
              height="4"
              width="20"
              x="2"
              y="10"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>,
]
`;

exports[`Counter renders correctly with set counter 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-qc2p2d{border-radius:lg;overflow:hidden;background:blue.400;box-shadow:xl;margin-bottom:10px;padding:5px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;vertical-align:center;}",
      }
    }
    data-emotion="css qc2p2d"
  />,
  <div
    className="css-qc2p2d"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-57pqds{-webkit-flex:1;-ms-flex:1;flex:1;vertical-align:center;}",
        }
      }
      data-emotion="css 57pqds"
    />
    <div
      className="css-57pqds"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-mmcpkq{font-weight:semibold;text-transform:uppercase;color:white;}",
          }
        }
        data-emotion="css mmcpkq"
      />
      <p
        className="chakra-text css-mmcpkq"
      >
        counter
      </p>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1j71ma5{font-size:2xl;color:white;}",
          }
        }
        data-emotion="css 1j71ma5"
      />
      <p
        className="chakra-text css-1j71ma5"
      >
        1
      </p>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-eg4v4b{margin-top:1px;}",
        }
      }
      data-emotion="css eg4v4b"
    />
    <div
      className="css-eg4v4b"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-c32o75{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;position:relative;white-space:nowrap;vertical-align:middle;outline:2px solid transparent;outline-offset:2px;width:auto;color:white;background-color:blue.400;}.css-c32o75:hover,.css-c32o75[data-hover]{background:blue.300;}.css-c32o75:active,.css-c32o75[data-active]{background:blue.400;}",
          }
        }
        data-emotion="css c32o75"
      />
      <button
        className="chakra-button css-c32o75"
        onClick={[Function]}
        type="button"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-19svxgr{width:5;height:5px;display:inline-block;line-height:1em;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;color:currentColor;vertical-align:middle;}",
            }
          }
          data-emotion="css 19svxgr"
        />
        <svg
          className="chakra-icon css-19svxgr"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <g
            fill="currentColor"
          >
            <rect
              height="4"
              width="20"
              x="2"
              y="10"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>,
]
`;

exports[`Counter renders correctly without counter 1`] = `
<div
  className="css-0"
/>
`;
